plugins/besecure.rb

```diff
@@ -0,0 +1,310 @@
+      res = http.start {|http| http.request(req)}
+
+      unless res
+        print_error("#{@hostname} - Connection timed out")
+        return ''
+      end
+      
+      if @debug
+        print_status(res)
+      end
+
+      body = ''
+      begin
+        body = JSON.parse(res.body)
+      rescue JSON::ParserError
+        print_error("#{@hostname} - Unable to parse the response")
+        return ''
+      end
+  
+      if body['error']
+        print_error("#{@hostname} - An error occured:")
+        print_error(body)
+        return ''
+      end
+
+      data = body['data']
+      data.each do |item|
+        tbl << [ item['ID'], item['Name'], item['PrettyRange']]
+      end
+
+      # print_good(body)
+  
+      print_good("beSECURE list of reports")
+      print_line
+      print_line tbl.to_s
+      print_line
+    end
+
+    def cmd_besecure_report_download(*args)
+      if args?(args, 4)
+        req = Net::HTTP::Post.new('/json.cgi', initheader={'Host'=>@hostname})
+        format_file = args[1]
+        req.set_form_data({'apikey' => @apikey, 'primary' => 'vulnerabilities', 'secondary' => 'report', 'action' => 'getreport', 'network' => args[0], 'format' => format_file})
+
+        if @debug
+          print_status(req.body)
+        end
+
+        http = Net::HTTP::new(@hostname, 443)
+        http.use_ssl = true
+        res = http.start {|http| http.request(req)}
+
+        unless res
+          print_error("#{@hostname} - Connection timed out")
+          return ''
+        end
+        
+        if @debug
+          print_status(res)
+        end
+
+        body = ''
+        begin
+          body = JSON.parse(res.body)
+        rescue JSON::ParserError
+          print_error("#{@hostname} - Unable to parse the response")
+          return ''
+        end
+    
+        if body['error']
+          print_error("#{@hostname} - An error occured:")
+          print_error(body)
+          return ''
+        end
+
+        # if @debug
+        #   print_status(body)
+        # end
+
+        decompressed = ''
+        if format_file != 'json'
+          compressed_base64 = body["compresseddata"]
+          compressed = Base64.decode64(compressed_base64)
+          decompressed = Zlib::Inflate.inflate(compressed)
+        else
+          decompressed = body
+        end
+
+        if @debug
+          print_status(decompressed)
+        end
+
+        ::FileUtils.mkdir_p(args[2])
+        name = ::File.join(args[2], args[3])
+        print_status("Saving report to #{name}")
+        output = ::File.new(name, "w")
+        output.puts(decompressed)
+        output.close
+      else
+        print_status("Usage: besecure_report_download <network_id> <format_name> <path> <report_name>")
+      end
+    end
+
+    def cmd_besecure_report_import(*args)
+      if args?(args, 2)
+        tempfile = Tempfile.new('results')
+
+        cmd_besecure_report_download(args[0], 'nbe', File.dirname(tempfile) + "/", File.basename(tempfile) )
```

Comment:
I think `nbe` is hard coded here, instead of `args[1]` ?

```suggestion
        cmd_besecure_report_download(args[0], args[1], File.dirname(tempfile) + "/", File.basename(tempfile) )
```

Reply:
The only relevant for importing is 'nbe', as MSF doesn't know how to import other file types, this is why its hardcoded to 'nbe' here

Reply:
@nrathaus Metasploit can import XML files, as and more - the full list of detected formats is [here](https://github.com/rapid7/metasploit-framework/blob/6034f48e8fa35f943051c0be85109f0973254fc6/lib/msf/core/db_manager/import.rb#L225-L468)

Just cross-linking this related discussion from the main thread incase it was missed:

https://github.com/rapid7/metasploit-framework/pull/13234#issuecomment-627979792

Reply:
Just circling back to this, it seems like the `nbe` format wasn't being imported correctly due to not having the following the header:

https://github.com/rapid7/metasploit-framework/blob/6034f48e8fa35f943051c0be85109f0973254fc6/lib/msf/core/db_manager/import.rb#L442-L445

I've just found an old document saying that an nbe file can either start with `timestamps` or `results`:

https://gitlab.com/kalilinux/packages/openvas-scanner/blob/48bf45dae0ba5d34abf38df336a2a650b8913cc2/doc/nbe_file_format.txt

But, that also doesn't align with the format that gets downloaded from besecure:

```
192.168.x.xx|http (80/tcp)|xxx|ATTACK|CVE: 
192.168.x.xx|http (80/tcp)|xxx|ATTACK|CVE: 
... etc ...
```

This seems like a bug in metasploit's file format detection either way. Potentially the nbe format has changed over time. Unfortunately I couldn't find any recent documentation for it. If you have any up to date knowledge on this, that would be a great help to get this landed :+1:

Reply:
We need to amend this bug on our end, I have opened a bug report on our system to get the 'scan_start' line

Reply:
The bug has been fixed from 10.5.17 and up of our beSECURE system

You can try it with the server you currently have access from tomorrow (Monday 18th)